### PR TITLE
feat: use custom aria labels for inputs with move announcements

### PR DESCRIPTION
**Pull request by mikeharv**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/RaspberryPiFoundation/blockly/issues/9836

### Proposed Changes

This PR allows move announcements to use `Input.ariaLabelProvider` when disambiguating connection candidates. 

image

If a provider is not provided, or one returns an empty string, we will fallback on the computed label using either the field labels for the input, or, if it's a statement input, a subset of inputs.

image

### Test Coverage

A new test ensures that a move announcement `disambiguates with custom input labels`. 

I also updated a few related tests to assert using string literals rather than duplicate computation logic that lived in the test file. This should make the tests easier to read and more likely to catch accidental regressions.

### CODE REVIEW

File: packages/blockly/core/block_aria_composer.ts [hunk 1]
```diff
@@ -205,11 +205,7 @@ export function getInputLabels(
 ): string[] {
   return block.inputList
     .filter((input) => input.isVisible())
-    .map((input) =>
-      input.getAriaLabelText() !== null
-        ? input.getAriaLabelText()!
-        : input.getLabel(verbosity),
-    );
+    .map((input) => input.getAriaLabelText() ?? input.getLabel(verbosity));
```
Comment (mikeharv) [inline packages/blockly/core/block_aria_composer.ts:208]: This doesn't change anything functionally, but using nullish is safe because it expresses the intended “use this unless it's null” logic directly, while avoiding both a duplicate method call and an unnecessary non-null assertion.

Comment (lizschwab) [reply packages/blockly/core/block_aria_composer.ts:208]: Interesting, I was under the impression that nullish coalescing was unsafe here because empty string evaluates to falsy but looks like the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) says that's not the case. Thanks, I wanted to do that to begin with. :D

**Review by lizschwab — APPROVED** [review]
lgtm
